File: 2. Week 2 - Scatter Plot/solutions/python/visualization.py
import pandas as pd
import matplotlib.pyplot as plt
import os

def visualize():
    data_path = '../../data/distance_vs_pace.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    df = pd.read_csv(data_path)
    
    # Plot
    plt.figure(figsize=(10, 6))
    
    # Scatter plot with color mapping
    sc = plt.scatter(df['Distance_km'], df['AveragePace_min_km'], 
                     c=df['AveragePace_min_km'], cmap='Purples', alpha=0.7)
    
    plt.colorbar(sc, label='Pace (min/km)')
    plt.title('Distance vs. Pace', fontsize=16)
    plt.xlabel('Distance (km)', fontsize=12)
    plt.ylabel('Pace (min/km)', fontsize=12)
    plt.grid(True, linestyle='--', alpha=0.5)
    
    plt.tight_layout()
    plt.savefig('distance_vs_pace_chart.png')
    print("Chart saved to distance_vs_pace_chart.png")
    plt.show()

if __name__ == "__main__":
    visualize()

File: 2. Week 2 - Scatter Plot/solutions/R/visualization.R
# Visualize Distance vs Pace using ggplot2

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

data_path <- "../../data/distance_vs_pace.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # Plot
  p <- ggplot(df, aes(x = Distance_km, y = AveragePace_min_km, color = AveragePace_min_km)) +
    geom_point(alpha = 0.7) +
    
    # Scale colors (Purple gradient)
    scale_color_gradient(low = "plum", high = "rebeccapurple") +
    
    # Labels
    labs(
      title = "Distance vs. Pace",
      x = "Distance (km)",
      y = "Pace (min/km)",
      color = "Pace"
    ) +
    
    # Theme
    theme_light() +
    theme(legend.position = "right")
  
  ggsave("distance_vs_pace_chart.png", plot = p, width = 10, height = 6)
  print("Chart saved to distance_vs_pace_chart.png")

} else {
  print(paste("Error: File not found at", data_path))
}
